crates/goose/src/agents/recipe_tools/dynamic_task_tools.rs

```diff
@@ -93,7 +94,7 @@ pub struct TaskParameter {
 
 pub fn should_enabled_subagents(model_name: &str) -> bool {
     let config = crate::config::Config::global();
-    let is_autonomous = config.get_param("GOOSE_MODE").unwrap_or("auto".to_string()) == "auto";
+    let is_autonomous = config.get_goose_mode().unwrap_or(GooseMode::Auto) == GooseMode::Auto;
```

Comment:
here and elsewhere; having a typed get_goose_mode is of course so much better, but should we add a default value to the pattern? we never read goose mode without defaulting to auto if it is not set